planner/core/plan.go

```diff
@@ -408,6 +411,10 @@ type basePhysicalPlan struct {
 	self             PhysicalPlan
 	children         []PhysicalPlan
 	cost             float64
+
+	// used by the new cost interface
+	planCostInit bool
+	planCost     float64
```

Comment:
According to the former comment, `costIsInit`, `cost` is enough?

Reply:
Emm, we cannot reuse the existing field `cost` here since it is being used by the old cost interface `plan.SetCost()/GetCost()`. In the next few PRs, I'll add some tests to compare costs calculated by the old interface and the new one, so we have to Introduce a new field `planCost` to store our new cost instead of resuing `cost`.